core-bundle/tests/EventListener/DataContainer/DefaultOperationsListenerTest.php

```diff
@@ -44,6 +44,7 @@ protected function tearDown(): void
 
     public function testAddsDefaultOperations(): void
     {
+        /** @var array $GLOBALS (signals PHPStan that the value is not immutable) */
```

Comment:
I think this is technically not 100% accurate.
It is more about that the inferred type (the [array shape](https://phpstan.org/writing-php-code/phpdoc-types#array-shapes) of `$GLOBALS`) changes outside of this scope, not the value itself.

```suggestion
        /** @var array $GLOBALS (signals PHPStan that the array shape may change) */
```

```suggestion
        /** @var array $GLOBALS (signals PHPStan that the type can not be inferred) */
```

Not sure. I’m fine either way or also keeping it as is.

Reply:
Adjusted in 2e1dc26ad18c73089824c6ae9525b7579919b19c.